odoo/tools/misc.py

```diff
@@ -1530,3 +1532,12 @@ def traverse_containers(val, type_):
     elif isinstance(val, collections.abc.Sequence):
         for v in val:
             yield from traverse_containers(v, type_)
+
+
+def sign_data(env, data):
+    """
+    Sign data with the database secret.
+    """
+    secret = env['ir.config_parameter'].sudo().get_param('database.secret').encode('utf-8')
+    data = repr((env.cr.dbname, data)).encode('utf-8')
```

Comment:
Why do you add the dbname in the signed data? If you want to add a bit of uniqueness, the dbuuid is maybe better (tolerant to server migration).
Any anyway `repr` of a tuple is weird? String concat is easier.

Reply:
@mart-e It's to be able to give list, string, integer, etc...
But ok for the dbuid :)

Reply:
But why would you want to do authenticate a list? The classic flow (also what you have in mass_mailing) is to compare a token with a single value. Ensure that the token belongs to the document you are currently viewing for instance. Don't overingeneer it.